crates/net/network/src/transactions/validation.rs

```diff
@@ -0,0 +1,555 @@
+//! Validation of [`NewPooledTransactionHashes68`] entries. These are of the form
+//! `(ty, hash, size)`. Validation and filtering of entries is network dependent. [``]
+
+use core::fmt;
+use std::{collections::HashMap, mem};
+
+use derive_more::{Deref, DerefMut, Display};
+use itertools::izip;
+use reth_eth_wire::{
+    NewPooledTransactionHashes66, NewPooledTransactionHashes68, ValidAnnouncementData,
+};
+use reth_primitives::{Signature, TxHash, TxType};
+use tracing::{debug, trace};
+
+/// The size of a decoded signature in bytes.
+pub const SIGNATURE_DECODED_SIZE_BYTES: usize = mem::size_of::<Signature>();
+
+/// Max encoded length of transaction types pre-blob transaction type.
+pub const MAX_ENCODED_PRE_BLOB_TX_LENGTH: usize = usize::pow(2, 17);
+
+/// Interface for validating a `(ty, size, hash)` tuple from a [`NewPooledTransactionHashes68`].
+pub trait ValidateTx68 {
+    /// Validates a [`NewPooledTransactionHashes68`] entry. Returns [`ValidationOutcome`] which
+    /// signals to the caller wether to fetch the transaction or wether to drop it, and wether the
+    /// sender of the announcement should be penalized.
+    fn should_fetch(&self, ty: u8, hash: TxHash, size: usize) -> ValidationOutcome;
+
+    /// Returns the reasonable maximum encoded transaction length configured for this network, if
+    /// any. This property is not spec'ed out but can be inferred by looking how much data can be
+    /// packed into a transaction for any given transaction type.
+    fn max_encoded_tx_length(&self, ty: TxType) -> Option<usize>;
+
+    /// Returns the strict maximum encoded transaction length for the given transaction type, if
+    /// any.
+    fn strict_max_encoded_tx_length(&self, tx: TxType) -> Option<usize>;
+
+    /// Returns the reasonable minimum encoded transaction length, if any. This property is not
+    /// spec'ed out but can be inferred by looking at which
+    /// [`reth_eth_wire::transactions::PooledTransactionsElement`] will successfully pass decoding
+    /// for any given transaction type.
+    fn min_encoded_tx_length(&self, ty: TxType) -> Option<usize>;
+
+    /// Returns the strict minimum encoded transaction length for the given transaction type, if
+    /// any.
+    fn strict_min_encoded_tx_length(&self, tx: TxType) -> Option<usize>;
```

Comment:
while writing docs...I realised we need to make new release if the blob tx size changes...that's not ideal. possibly why you have been getting an uneasy feeling about the encoded length bounds @mattsse ?

- moved the encoded length limits from `TxType`, to be configured on a network basis in validation.
- introduced 'strict limit' and 'limit'. this way we can get more interesting debug output about malformed txns, that might just be slightly too big or too small. I refer to this as "reasonably too small/big" in the log messages rather than "invalid". we get this interesting debug output without compromising flexibility.

let me know what you think before I fix the filter tests. @mattsse @Rjected

Reply:
I like this approach! this does not leak into the txtype